os/user/getent_cgo.go

```diff
@@ -0,0 +1,149 @@
+// +build aix dragonfly freebsd !android,linux netbsd openbsd solaris darwin
+// +build cgo,!osusergo
+
+package user
+
+/*
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <grp.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+
+static void resetErrno(){
+	errno = 0;
+}
+*/
+import "C"
+
+// iterateUsersHelper is helper structure, which is primarily used to distinguish between testing and normal iterateUsers
+// functionality. Since we want to have a hermetic and reproducible testing environment, we can't rely on users
+// available in the system. As such, we will use fgetpwent library call along with testdata/users.txt fixture when
+// running tests for this package
+type iterateUsersHelper struct {
```

Comment:
this helper is OK for test code, but it pollutes the production code with test-only helpers.

Can we think of another approach when the "real" code uses getpwent, but we make the test go through fgetpwent?

Test only, where we assume `stream` is a variable in scope where this macro is used:
```
#define __getpwent fgetpwent(stream)
```

And prod:
```
#define __getpwent getpwent
```

Or any other approaches?